src/libs/snapshots.c

```diff
@@ -668,6 +683,21 @@ static int lua_take_snapshot(lua_State *L)
   return 0;
 }
 
+static int lua_clear_snapshots(lua_State *L)
+{
+  dt_lib_module_t *self = lua_touserdata(L, lua_upvalueindex(1));
+  dt_lib_snapshots_t *d = (dt_lib_snapshots_t *)self->data;
```

Comment:
from here, why not call `_internal_gui_reset(self);` The only diff is the redraw center, but well, maybe its needed when called from Lua ?

Reply:
Oops, _internal_gui_reset shouldn't be there.  My first cut at it was copy `gui_reset` to `_internal_gui_reset` and calling `_internal_gui_reset` from `gui_reset` and `clear_snapshots`.  When I tested it appeared that I got a screen flash when I loaded the image and I figured it was the `dt_control_queue_redraw_center` causing it.  Let me do some more testing and see what I can come up with.

Reply:
Got rid of `_internal_gui_reset`.  Changed `lua_clear_snapshots` to just use `gui_reset`.  Tested and didn't see any screen flashes.